qiskit_nature/algorithms/ground_state_solvers/minimum_eigensolver_factories/vqe_ucc_factory.py

```diff
@@ -81,107 +86,167 @@ def __init__(
                 ansatz, the evaluated mean and the evaluated standard deviation.`
             kwargs: any additional keyword arguments will be passed on to the VQE.
         """
-        self.quantum_instance = quantum_instance
-        self.optimizer = optimizer
-        self.initial_point = initial_point
-        self.gradient = gradient
-        self.expectation = expectation
-        self.include_custom = include_custom
-        self.ansatz = ansatz
-        self.initial_state = initial_state
-        self.callback = callback
+        self._ansatz = ansatz
+        self._initial_state = initial_state
+
+        # Not needed
+        self._quantum_instance = quantum_instance
+        self._optimizer = optimizer
+        self._initial_point = initial_point
+        self._gradient = gradient
+        self._expectation = expectation
+        self._include_custom = include_custom
+        self._callback = callback
         self._kwargs = kwargs
 
-    @property
+        self._vqe = VQE(
+            quantum_instance=quantum_instance,
+            optimizer=optimizer,
+            initial_point=initial_point,
+            gradient=gradient,
+            expectation=expectation,
+            include_custom=include_custom,
+            callback=callback,
+            **kwargs,
+        )
+
+    @property  # type: ignore
+    @deprecate_property(
+        "0.4", additional_msg="Use `minimum_eigensolver` and 'solver properties' instead."
```

Comment:
```suggestion
        "0.4", additional_msg="Use `minimum_eigensolver` and 'solver properties' instead"
```
The dot gets added by the decorator